Dockerfile-nanoserver.template

```diff
@@ -0,0 +1,52 @@
+FROM microsoft/windowsservercore as download
+
+SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]
+
+ENV NODE_VERSION 0.0.0
+
+ENV GPG_VERSION 2.3.4
+
+RUN [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.SecurityProtocolType]::Tls12 ; \
+    Invoke-WebRequest $('https://files.gpg4win.org/gpg4win-vanilla-{0}.exe' -f $env:GPG_VERSION) -OutFile 'gpg4win.exe' -UseBasicParsing ; \
+    Start-Process .\gpg4win.exe -ArgumentList '/S' -NoNewWindow -Wait ;
+
+RUN @( \
+      "${NODE_KEYS[@]}"
+    ) | foreach { \
+      gpg --keyserver ha.pool.sks-keyservers.net --recv-keys $_ ; \
+    } ; \
+    [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.SecurityProtocolType]::Tls12 ; \
+    Invoke-WebRequest $('https://nodejs.org/dist/v{0}/SHASUMS256.txt.asc' -f $env:NODE_VERSION) -OutFile 'SHASUMS256.txt.asc' -UseBasicParsing ; \
+    gpg --batch --decrypt --output SHASUMS256.txt SHASUMS256.txt.asc ; \
+    Invoke-WebRequest $('https://nodejs.org/dist/v{0}/node-v{0}-win-x64.zip' -f $env:NODE_VERSION) -OutFile 'node.zip' -UseBasicParsing ; \
+    $sum = $(cat SHASUMS256.txt.asc | sls $('  node-v{0}-win-x64.zip' -f $env:NODE_VERSION)) -Split ' ' ; \
+    if ((Get-FileHash node.zip -Algorithm sha256).Hash -ne $sum[0]) { Write-Error 'SHA256 mismatch' } ; \
+    Expand-Archive node.zip -DestinationPath C:\ ; \
+    Rename-Item -Path $('C:\node-v{0}-win-x64' -f $env:NODE_VERSION) -NewName 'C:\nodejs'
+
+ENV YARN_VERSION 0.0.0
+
+RUN [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.SecurityProtocolType]::Tls12 ; \
+    Invoke-WebRequest $('https://yarnpkg.com/downloads/{0}/yarn-{0}.msi' -f $env:YARN_VERSION) -OutFile yarn.msi -UseBasicParsing ; \
+    $sig = Get-AuthenticodeSignature -filepath yarn.msi ; \
+    if ($sig.Status -ne 'Valid') { Write-Error 'Authenticode signature is not valid' } ; \
+    if (@( \
+      'AF764E1EA56C762617BDC757C8B0F3780A0CF5F9' \
+      ) -notcontains $sig.SignerCertificate.Thumbprint) { Write-Error 'Unknown signer certificate' } ; \
+    Start-Process msiexec.exe -ArgumentList '/i', 'yarn.msi', '/quiet', '/norestart' -NoNewWindow -Wait
+
+FROM microsoft/nanoserver
+
+SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]
+
+ENV NODE_VERSION 0.0.0
+ENV YARN_VERSION 0.0.0
+
+COPY --from=download /nodejs /nodejs
+COPY --from=download [ "/Program Files (x86)/yarn", "/yarn" ]
+
+RUN New-Item $($env:APPDATA + '\npm') -Type Directory ; \
```

Comment:
The nanoserver images beginning with 1709 do not have PowerShell installed.
That's why I've switched to  `setx` command https://github.com/StefanScherer/dockerfiles-windows/blob/66ad7a25005ba5926829e145592983b498c2a6cb/node/10.0/nano/Dockerfile#L54

But also beware that beginning with 1709 the default user is no longer the ContainerAdministrator.
Therefore you cannot use `setx /M` to write to HKLM.

I use these three lines https://github.com/StefanScherer/dockerfiles-windows/blob/66ad7a25005ba5926829e145592983b498c2a6cb/node/10.0/nano/Dockerfile.1709#L54-L56 as a workaround

```Dockerfile
USER ContainerAdministrator
RUN setx /m PATH "%PATH%;C:\nodejs;C:\yarn\bin"
USER ContainerUser
```

to set the PATH for every user. But Microsoft plans to get rid of the ContainerAdministrator completely, but I do not have more details about of the why and how we can prepare registry keys in HKLM in another way then.

Setting path only with `setx` without `/M` means that you cannot run the container with yet another user you create in a container. This other user then does not have the PATH set for Node.js and Yarn.